src/nodes.coffee

```diff
@@ -2684,11 +2687,13 @@ exports.Class = class Class extends Base
   # initializer as an escape hatch for ES features that are not implemented
   # (e.g. getters and setters defined via the `get` and `set` keywords as
   # opposed to the `Object.defineProperty` method).
-  addInitializerExpression: (node) ->
+  addInitializerExpression: (node, o) ->
     if node.unwrapAll() instanceof PassthroughLiteral
       node
     else if @validInitializerMethod node
       @addInitializerMethod node
+    else if not o.compiling and @validClassProperty node
+      @addClassProperty node
```

Comment:
In Babel, static class properties generate `ClassProperty` AST nodes, so if we're generating AST, create instances of the new `ClassProperty` node class as we encounter static properties while walking the class body

Reply:
I’m not sure about `o.compiling`. This appears to be the only place you need it; are you using it elsewhere in your Prettier branch for other things?

It seems like you’re using this in order to create a different node tree for JS versus AST generation. I’m wondering, why should we have separate trees? That seems like it would just make the overall codebase harder to maintain.

If you need a new child class for class properties or static class properties, can you make that new class used for both JS and AST? I know that might mean a bigger refactor on the JS side, but I think there’s value in keeping the two node trees/structures the same.

Reply:
> are you using it elsewhere in your Prettier branch for other things?

Yes, and it may come in handy elsewhere in these AST PRs

> I’m wondering, why should we have separate trees? That seems like it would just make the overall codebase harder to maintain

Well, let's look at this from the perspective of what we've been doing so far. In some places we've introduced node classes that also get used during compile-to-JS (eg the new JSX node classes). In other places we've introduced node classes that only get used during AST generation (eg `ObjectProperty`). We've been basing those decisions on things like how straightforward it is to integrate the new node classes

So basically yes there's a convenience to having the AST and JS node trees match but we've already set precedent (for good reasons) for having them diverge

The only difference here is that that divergence is taking place inside a shared helper method (rather than eg an AST-specific method) - sharing helper methods between compile-to-JS and AST generation is good for similar reasons (more maintainable, DRYer)

And conceptually I don't have a problem with a flag on `o` indicating whether we're in "compile-to-JS" mode or "AST generation" mode - `o` is basically a context object for various things about keeping track of the current task (whether that's compiling to JS or AST generation), so differentiating those two tasks seems like part of that context

> If you need a new child class for class properties or static class properties, can you make that new class used for both JS and AST?

So in terms of this specific choice (not to try and reuse `ClassProperty` in compile-to-JS), I avoided it because class compilation is a rather complex process and I felt like it wasn't worth messing with its nuts and bolts (just to avoid an AST-specific node class)